### DISCUSSION TITLE
Issue with pushing large file to GitHub

### DISCUSSION THREAD

**Discussion by valexsyu** [community/community#56630] [Pull Requests] [answered]
### Select Topic Area

General

### Body

I accidentally added and committed a file of size 633.39MB using git add and git commit. As a result, I encountered issues when trying to push the changes using git push. I then removed the 633.39MB file, but I still encountered the following error when attempting to push:

Total 41 (delta 28), reused 1 (delta 0)
remote: Resolving deltas: 100% (28/28), completed with 22 local objects.
remote: error: Trace: 4c8cf6a6bb87835bd62341f0a5f8f81c6353a3335c76ddcab5ddaf61f4eb1aa1
remote: error: See https://gh.io/lfs for more information.
remote: error: File pytorch_model.bin is 633.39 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/valexsyu/Reorder_nat.git
 ! [remote rejected]   main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/valexsyu/Reorder_nat.git'

I followed the steps mentioned in the documentation to use Git Large File Storage (LFS) and configured it properly. However, I'm still encountering the same error.
I would appreciate any assistance in resolving this issue.

Thank you.

**Comment by airtower-luna** [comment]
> I accidentally added and committed a file of size 633.39MB using git add and git commit. As a result, I encountered issues when trying to push the changes using git push. I then removed the 633.39MB file,

How did you remove the file? If you did `git rm` and make another commit, the file is still part of the history. You'll need to use `git rebase -i` to change the repository history and fix the commit that added the file. See [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) for details.

If you don't actually want to version that file you don't need LFS at all. :slightly_smiling_face:

**Reply by valexsyu** [reply, marked as answer]
Thanks for your reply, I used the git reset --soft HEAD^ and solved the issue.

**Reply by Cathya-21** [reply]
what are the steps to follow because I encountered the same problem as you?

**Reply by Cathya-21** [reply]
@valexsyu

**Reply by garincsr** [reply]
@valexsyu Please give the steps to follow

**Reply by Raunak1818** [reply]
where is answer? @valexsyu

**Comment by pasde1** [comment]
When encountering issues pushing large files to GitHub, it's important to understand that there are limitations on file size and storage capacity. GitHub has certain restrictions in place to ensure efficient and reliable code collaboration. The maximum file size limit on GitHub is 100 MB. However, it's generally recommended to keep file sizes smaller to optimize performance and ease collaboration.

If you're facing difficulties pushing large files, [here](https://trafficridermodapk.org/traffic-rider-apk-for-pc/) are a few potential solutions:

Git Large File Storage (Git LFS): Git LFS is an extension that GitHub provides to handle large files more effectively. It replaces large files in your repository with text pointers while storing the actual file content in a separate location. This helps manage version control and reduces the overall repository size. You can install Git LFS and configure it to handle large files in your repository.

Exclude or compress large files: If the large files in question are not essential for the functioning of your repository, consider excluding them from version control using a .gitignore file. Alternatively, you can compress the files or split them into smaller parts before pushing them to GitHub. This can help reduce the file size and make it more manageable.

Use Git submodules or external storage: If the large files are not directly related to the codebase but are necessary for your project, you can consider using Git submodules or storing the files externally. With submodules, you can include a reference to another repository that contains the large files. Alternatively, you can store the files on a separate storage service (such as AWS S3 or Google Drive) and include links or instructions for accessing those files in your repository.

GitHub Releases or Git tags: Instead of including large files directly in your repository, you can create a release or Git tag on GitHub and upload the large files as release assets. This way, you can provide the necessary files as separate downloads without bloating the repository size.

Remember, it's essential to be mindful of file sizes and ensure that your repository remains efficient and accessible to collaborators. If the large files are not critical to the repository itself, it's often better to find alternative methods for sharing or storing them.

If you continue to experience issues pushing large files or need more specific guidance, it may be helpful to consult GitHub's documentation or reach out to their support for further assistance.

**Comment by pasde1** [comment]
Pushing large files to GitHub can sometimes be a challenge due to file size limitations and potential performance issues. GitHub has a recommended file size limit of 100 MB for individual files. If you encounter issues while pushing large files, [here ](https://mistplaymodapk.com/)are some possible solutions:

Git Large File Storage (Git LFS): Git LFS is an extension that replaces large files in your repository with text pointers, storing the actual file contents on a separate server. It allows you to push and pull large files more efficiently. To use Git LFS, you need to install it and configure it for your repository.

Ignore Large Files: If the large files are not crucial for the project's history or development, consider adding them to your repository's .gitignore file. This prevents Git from tracking and including them during commits, thus avoiding the push issue.

Git Alternatives: If the large files are not suitable for inclusion in your Git repository, consider hosting them separately. You can use cloud storage services like Amazon S3, Google Drive, or Dropbox to store the files and provide download links or instructions in your repository's README file.

Archive or Compress Large Files: If the large files are necessary for your project but exceed GitHub's file size limit, consider archiving or compressing them into smaller files. You can use tools like ZIP, RAR, or 7z to create compressed archives before pushing them to your repository. However, note that this approach may not be suitable for binary files like images or videos that are already compressed.

Git Submodules: If the large files are shared across multiple repositories or projects, you can consider using Git submodules. With sub

**Comment by Qureshi2525** [comment]
remote: error: Trace: 04d83a704f32c451a3089bdf2e637a9c9dc55aa44dff7d02bbf96d2149d46a51
remote: error: See https://gh.io/lfs for more information.
remote: error: File vedio/devathon.mp4 is 106.75 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File S

**Comment by lesimoes** [comment]
Try this

```
git filter-branch --force --index-filter \            19s 11:21:05
  'git rm --cached --ignore-unmatch PATH/FILE.EXE' \
  --prune-empty --tag-name-filter cat -- --all
```

Then

`git push origin BRANCH --force --all`

source: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository
